nix/pkgs/openebs/cargo-project.nix

```diff
@@ -0,0 +1,162 @@
+    "mayastor/dependencies/control-plane/utils/dependencies/tracing-filter"
+    "mayastor/dependencies/control-plane/utils/dependencies/version-info"
+    "mayastor/dependencies/control-plane/utils/utils-lib"
+    "mayastor/dependencies/control-plane/utils/hyper-body"
+    "mayastor/dependencies/control-plane/utils/shutdown"
+    "mayastor/dependencies/control-plane/utils/platform"
+    "mayastor/dependencies/control-plane/utils/pstor"
+    "mayastor/dependencies/control-plane/rpc"
+    "mayastor/dependencies/control-plane/k8s/forward"
+    "mayastor/dependencies/control-plane/k8s/proxy"
+    "mayastor/k8s"
+    "mayastor/console-logger"
+    "mayastor/constants"
+    "mayastor/dependencies/control-plane/k8s/operators"
+  ];
+  src = sourcer.whitelistSource ../../../. src_list;
+  static_ssl = (pkgs.pkgsStatic.openssl.override {
+    static = true;
+  });
+  hostTarget = pkgs.rust.toRustTargetSpec pkgs.hostPlatform;
+  buildProps = rec {
+    name = "extensions-${version}";
+    inherit version src;
+    GIT_VERSION_LONG = "${gitVersions.long}";
+    GIT_VERSION = "${gitVersions.tag_or_long}";
+
+    nativeBuildInputs = [ clang pkg-config git paperclip which protobuf ];
+    buildInputs = [ llvmPackages.libclang openssl utillinux ];
+    doCheck = false;
+  };
+  release_build = { "release" = true; "debug" = false; };
+  flags =
+    if builtins.stringLength rustFlags > 0
+    then builtins.split " " rustFlags
+    else if static
+    then [ "-C" "target-feature=+crt-static" ]
+    else [ ];
+in
+let
+  build_with_naersk = { buildType, cargoBuildFlags }:
+    naersk.buildPackage (buildProps // {
+      release = release_build.${buildType};
+      cargoBuildOptions = attrs: attrs ++ cargoBuildFlags;
+      preBuild = ''
+        # don't run during the dependency build phase
+        if [ ! -f build.rs ]; then
+          patchShebangs .mayastor/dependencies/control-plane/scripts/rust/
+          .mayastor/dependencies/control-plane/scripts/rust/generate-openapi-bindings.sh --skip-git-diff
+        fi
+      '';
+      doCheck = false;
+      usePureFromTOML = true;
+    });
+  build_with_default = { buildType, cargoBuildFlags }:
+    rustPlatform.buildRustPackage (buildProps // {
+      inherit buildType cargoBuildFlags;
+      preBuild = ''
+        patchShebangs .mayastor/dependencies/control-plane/scripts/rust/
+      '' + pkgs.lib.optionalString (static) ''
+        # the rust builder from nixpkgks does not parse target and just uses the host target...
+        export NIX_CC_WRAPPER_TARGET_HOST_${builtins.replaceStrings [ "-" ] [ "_" ] hostTarget}=
+        export OPENSSL_STATIC=1
+        export OPENSSL_LIB_DIR=${static_ssl.out}/lib
+        export OPENSSL_INCLUDE_DIR=${static_ssl.dev}/include
+      '';
+      ${if flags == [ ] then null else "RUSTFLAGS"} = flags;
+      cargoLock = {
+        lockFile = ./Cargo.lock;
+      };
+    });
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
```

Comment:
Is this path correct?

Reply:
made it to relative path from this script.